distribution/lib/Standard/Base/0.0.0-dev/src/Data/Time/Period.enso

```diff
@@ -0,0 +1,146 @@
+from Standard.Base import all
+
+import Standard.Base.Data.Time.Duration
+
+polyglot java import java.time.Period as Java_Period
+polyglot java import java.time.DateTimeException
+polyglot java import java.lang.ArithmeticException
+
+
+## Create a Period representing the time interval between two dates.
+
+   Arguments:
+   - start_date_inclusive: The start date of the period, included.
+   - end_date_exclusive: The end date of the period, excluded.
+
+   > Example
+     Get a Period between 2022-10-21 and 2022-09-12
+
+         import Standard.Base.Data.Time.Period
+
+         example_period = Period.between (Date.new 2022 10 21) (Date.new 2022 9 12)
+between : Date -> Date -> Period
+between start_date_inclusive end_date_exclusive =
+    Period.Period_Data (Java_Period.between start_date_inclusive end_date_exclusive)
+
+## Create a new Period from years, months and days.
+
+   Arguments:
+   - years: Amount of years.
+   - months: Amount of months.
+   - days: Amount of days.
+
+   > Example
+     Create a Period of 2 years and 5 days
+
+         import Standard.Base.Data.Time.Period
+
+         example_period = Period.new 2 0 5
+new : Integer -> Integer -> Integer -> Period
+new years=0 months=0 days=0 =
```

Comment:
I'm wondering if it would make sense to create Period module constructor functions like `days`, `years` and `months` in addition to `new`. I think one will rarely set more than one of those parameters and it would seem simpler, as can be seen in the tests.
It's a minor question to @jdunkerley and @radeusgd, though.

Reply:
See https://github.com/enso-org/enso/pull/3818#discussion_r1003542930 - in a nutshell, the problem is in type clarity: `1.hours` results in `Duration`, whereas `1.days` results in `Period`, which seems confusing. Therefore, with @radeusgd , we decided to get rid of all these extension methods altogether.

Reply:
> From how I understood @hubertp's comment, the idea is to instead provide constructors like Period.days 5, Period.years 2 etc.

That seems pretty useful. Also for `Duration`. Having, e.g., `(Duration.hours 3) + (Duration.minutes 20)` seems much better than `(Duration.new hours=3) + (Duration.new minutes=20`.

Reply:
Yeap, my idea was to have `Period.hours 2` rather than having to write `Period.new hours=2` every time.

Reply:
Added shorthand constructors for both `Period` and `Duration`.